translib/transformer/xfmr_intf.go

```diff
@@ -912,10 +994,44 @@ var intf_pre_xfmr PreXfmrFunc = func(inParams XfmrParams) error {
 
 var intf_subintfs_table_xfmr TableXfmrFunc = func(inParams XfmrParams) ([]string, error) {
 	var tblList []string
-
 	pathInfo := NewPathInfo(inParams.uri)
+	ifName := pathInfo.Var("name")
 	idx := pathInfo.Var("index")
 
+	if inParams.oper == SUBSCRIBE {
+		var _intfTypeList []E_InterfaceType
+
+		if idx == "*" || idx != "0" {
+			err_str := "Subinterfaces not supported"
+			return tblList, tlerr.NotSupported(err_str)
+		}
+
+		if ifName == "*" {
+			_intfTypeList = append(_intfTypeList, IntfTypeEthernet)
+		} else {
+			_ifType, _, _err := getIntfTypeByName(ifName)
+			if _ifType == IntfTypeUnset || _err != nil {
+				return tblList, errors.New("Invalid interface type IntfTypeUnset")
+			}
+			_intfTypeList = append(_intfTypeList, _ifType)
+		}
+
+		for _, _ifType := range _intfTypeList {
+			_intfTblName, _ := getIntfTableNameByDBId(IntfTypeTblMap[_ifType], inParams.curDb)
+			tblList = append(tblList, _intfTblName)
+		}
+
+		log.V(3).Info("intf_subintfs_table_xfmr: URI: ", inParams.uri, " OP:", inParams.oper, " ifName:", ifName, " idx:", idx, " tblList:", tblList)
+		return tblList, nil
+	}
+
+	intfType, _, ierr := getIntfTypeByName(ifName)
```

Comment:
Do we really need this `getIntfTypeByName` code here? because subscribe code returns with above if-check. And this check was not there earlier in previous PR for GET/SET support. This question might come from external review. We should try to remove this code and test how things are.